### DISCUSSION TITLE
Is there any idea for supporting s3 ACL

### DISCUSSION THREAD

**Discussion by shichanglin5** [seaweedfs/seaweedfs#3319] [General]
In some usage scenarios, some objects of a bucket can be publicly downloaded for external personnel, but other objects are accessed internally. It will be very convenient if acl is supported;

An initial idea：
at `weed/server/filer_server_handlers_write_autochunk.go:242`，save the acl config  when put object
```go
entry.Extended = SaveAmzMetaData(r, entry.Extended, false)
```

at `weed/s3api/auth_credentials.go:185`, If bucket acls are enabled, object metadata may need to be obtained to determine if public reads are allowed
```go
func (iam *IdentityAccessManagement) authRequest(r *http.Request, action Action) (*Identity, s3err.ErrorCode)
```

The above is just the most basic idea. Enabling object acl means that when executing `authRequest`, may need to execute `getEntry` to determine acl

**Comment by kmlebedev** [comment]
At the first stage, you can support default ACLs
https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html
